pkg/metrics/exporters/stackdriver/stackdriver.go

```diff
@@ -31,26 +40,37 @@ func Start(ctx context.Context) error {
 		return err
 	}
 
-	exporter, err := stackdriver.NewExporter(stackdriver.Options{
-		MetricPrefix:      metricPrefix,
-		MonitoredResource: monitoredresource.Autodetect(),
-	})
+	e, err := stackdriver.New(stackdriver.WithMetricDescriptorTypeFormatter(func(desc metricdata.Metrics) string {
+		return fmt.Sprintf("%s/%s", metricPrefix, desc.Name)
+	}))
 	if err != nil {
 		if *ignoreMissingCreds {
 			log.Error(err, "Error initializing stackdriver exporter, not exporting stackdriver metrics")
 			return nil
 		}
 		return err
 	}
-
-	if err := exporter.StartMetricsExporter(); err != nil {
-		if *ignoreMissingCreds {
-			log.Error(err, "Error starting stackdriver exporter, not exporting stackdriver metrics")
-			return nil
-		}
+	awsResource, err := ec2.NewResourceDetector().Detect(ctx)
+	if err != nil {
 		return err
 	}
-	defer exporter.StopMetricsExporter()
+	resource := awsResource
+	gcpResource, err := gcp.NewDetector().Detect(ctx)
+	if err != nil {
+		return err
+	}
+	if gcpResource != nil {
+		resource = gcpResource
+	}
```

Comment:
I think this all can be simplified so as to pass in the detectors to a resource. 

Also, Something that the detector documentation for GCP did not make clear (at least to me) was that in order for the OTEL_RESOURCE_ATTRIBUTES to be parsed from the env we also need to set the Detector that would look at environment attributes. So putting those two things together could look something like

```
        ...
	res, err := resource.New(ctx,
		resource.WithDetectors(gcp.NewDetector(), ec2.NewResourceDetector()),
		resource.WithTelemetrySDK(),
		// this detector will unpack the OTEL_RESOURCE_ATTRIBUTES; for gcp, setting those attrs
		// will, in turn, detect the monitored resource as a "k8s_container".
		resource.WithFromEnv(), 
	)
        ...
```

Reply:
Updated the code